src/main/scala/io/phdata/retirementage/loadgen/LoadGenerator.scala

```diff
@@ -1,39 +1,194 @@
+//    // write dimension table to disk
+//    dimensionDf.write
+//      .mode(SaveMode.Overwrite)
+//      .saveAsTable(s"${conf.databaseName()}.${conf.dimName()}")
+//    // generate subdimension table
+//    val subDimensionDf = generateTableFromParent(spark,
+//                                                 conf.subDimensionCount(),
+//                                                 1,
+//                                                 conf.dimensionCount(),
+//                                                 dimensionDf)
+//    // write subdimension table to disk
+//    subDimensionDf.write
+//      .mode(SaveMode.Overwrite)
+//      .saveAsTable(s"${conf.databaseName()}.${conf.subName()}")
   }
 
+  // Creating a case class to use for generateTable
+  case class LoadGenTemp(id: String, payload: String, dimensionid: String, expirationdate: String)
+
   /**
     * Generates a test dataframe
     * @param numRecords Number of records in the dataframe
     * @param payloadBytes Payload size
     * @return The test dataframe
     */
-  def generateTable(numRecords: Int, payloadBytes: Int): DataFrame = ???
+  def generateTable(spark: SparkSession, numRecords: Int, payloadBytes: Int) = {
+
+    /**
+      * Create the following schema:
+      *   -- id: String (nullable = false)
+      *   -- payload: String (nullable = false)
+      *   -- dimensionId: String (nullable = false)
+      *   -- expirationDate: String (nullabe = false)
+      */
+    val schema = StructType(StructField("id", StringType, false) :: Nil)
+      .add(StructField("payload", StringType, false))
+      .add(StructField("dimensionid", StringType, false))
+      .add(StructField("expirationDate", StringType, false))
+    // Create a string with specified bytes
+    val byteString = "a" * payloadBytes
+
+    // Maximum number of records per partition
+    val maxPartitionRecords = 50000
+
+    // Calculating the number of partitions and the records per partition
+    val numPartitions       = math.ceil(numRecords.toDouble / maxPartitionRecords).toInt
+    val recordsPerPartition = math.ceil(numRecords.toDouble / numPartitions).toInt
+
+    val newdata = Range(1, recordsPerPartition).map(
+      x =>
+        Seq(UUID.randomUUID().toString(),
+            byteString,
+            UUID.randomUUID().toString(),
+            tempDateGenerator()))
+    val rows = newdata.map(x => Row(x: _*))
+    val rdd  = spark.sparkContext.makeRDD(rows)
+    val df   = spark.createDataFrame(rdd, schema)
+
+    val dfs = Range(1, numPartitions).map(x => df)
+
+    // Union the Sequence of DataFrames onto finalTable
+    val duplicateDf: DataFrame = dfs.reduce((l, r) => l.union(r))
+
+    import spark.implicits._
+    // Create temporary dataset to make changes on
+    val tempDs: Dataset[LoadGenTemp] = duplicateDf.as[LoadGenTemp]
+
+    // Create random data
+    val idRandomDs: Dataset[LoadGenTemp] = tempDs.map {
```

Comment:
Can all of these transformations be done in one call to `map`. Lots of unnecessary objects are created each time `copy` is called